From 3a84b6e3fe277b44127eae28068e96ae4c892be5 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 17 Nov 2022 12:52:47 +0200
Subject: [PATCH] [SW-106014] gaudi3: use arcboot boot_info get api

boot_info get api is exported by arcboot and available for ppboot
to use. Calling the arcboot api instead of the local implementation
will decrease the ppboot size. Ppboot should be kept as thin as it
can be, so boot time loading/processing/validating operations will
perform as fast as we can.

Change-Id: Iec3a0930e4094a93d50c348f1534b55b985574a9
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 4533c02..4d39023 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -565,7 +565,7 @@
 
 int ppboot_boot_info_get(u32 type, void *data, u32 max_size, u32 *copy_size)
 {
-	return boot_info_get(type, data, max_size, copy_size);
+	return boot_info_get_no_merge(type, data, max_size, copy_size);
 }
 
 #define CCM_TERMINATE_MASK	(D0_PIF_ARC_CCM_TERMINATE_D_EN_M | D0_PIF_ARC_CCM_TERMINATE_I_EN_M)
